pyccel/parser/semantic.py

```diff
@@ -1656,7 +1413,7 @@ def _visit(self, expr, **settings):
                 return obj
 
         # Unknown object, we raise an error.
-        errors.report(PYCCEL_RESTRICTION_TODO, symbol=type(expr),
+        return errors.report(PYCCEL_RESTRICTION_TODO, symbol=type(expr),
```

Comment:
Recurring question about the need for a `return`

Reply:
See above